File: i2f-springboot/i2f-springboot-secure-starter/src/main/java/i2f/springboot/secure/data/SecureHeader.java
package i2f.springboot.secure.data;

import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * @author Ice2Faith
 * @date 2023/6/13 10:11
 * @desc
 */
@Data
@NoArgsConstructor
public class SecureHeader {
    /**
     * 签名
     */
    public String sign;
    /**
     * 一次性消息
     */
    public String nonce;
    /**
     * 数字签名
     */
    public String digital;
    /**
     * 随机秘钥
     */
    public String randomKey;
    /**
     * 服务端非对称签名
     */
    public String serverAsymSign;

    public String clientAsymSign;
}

File: i2f-springboot/test-secure-starter/web/secure-web/src/secure/data/SecureHeader.js
/**
 * 请求头sswh的解码组成部分
 */
const SecureHeader = {
    // 签名
    sign: null,
    // 一次性消息
    nonce: null,
    // 数字签名
    digital: null,
    // 随机秘钥
    randomKey: null,
    // 服务端非对称签名
    serverAsymSign: null,
    // 客户端非对称签名
    clientAsymSign: null,
    newObj: () => {
        return {
            sign: null,
            nonce: null,
            digital: null,
            randomKey: null,
            serverAsymSign: null,
            clientAsymSign: null
        }
    }
}
export default SecureHeader
